src/number/n_transExt.jl

```diff
@@ -0,0 +1,360 @@
+#
+#   Powering
+#
+###############################################################################
+
+function ^(x::n_transExt, y::Int)
+    y < 0 && throw(DomainError())
+    if isone(x)
+       return x
+    elseif y == 0
+       return one(parent(x))
+    elseif y == 1
+       return x
+    else
+       p = libSingular.n_Power(x.ptr, y, parent(x).ptr)
+       return parent(x)(p)
+    end
+end
+
+###############################################################################
+#
+#   Exact division
+#
+###############################################################################
+
+function inv(x::n_transExt)
+   c = parent(x)
+   p = libSingular.n_Invers(x.ptr, c.ptr)
+   return c(p)
+end
+
+function divexact(x::n_transExt, y::n_transExt)
+   c = parent(x)
+   p = libSingular.n_Div(x.ptr, y.ptr, c.ptr)
+   return c(p)
+end
+
+###############################################################################
+#
+#   GCD and LCM
+#
+###############################################################################
+
+function gcd(x::n_transExt, y::n_transExt)
+   if x == 0 && y == 0
+      return zero(parent(x))
+   end
+   par = parent(x)
+   p = libSingular.n_Gcd(x.ptr, y.ptr, par.ptr)
+   return par(p)
+end
+
+###############################################################################
+#
+#   Unsafe functions
+#
+###############################################################################
+
+function addeq!(x::n_transExt, y::n_transExt)
+   x.ptr = libSingular.n_InpAdd(x.ptr, y.ptr, parent(x).ptr)
+   return x
+end
+
+function mul!(x::n_transExt, y::n_transExt, z::n_transExt)
+   ptr = libSingular.n_Mult(y.ptr, z.ptr, parent(x).ptr)
+   libSingular.n_Delete(x.ptr, parent(x).ptr)
+   x.ptr = ptr
+   return x
+end
+
+function add!(x::n_transExt, y::n_transExt, z::n_transExt)
+   ptr = libSingular.n_Add(y.ptr, z.ptr, parent(x).ptr)
+   libSingular.n_Delete(x.ptr, parent(x).ptr)
+   x.ptr = ptr
+   return x
+end
+
+function zero!(x::n_transExt)
+   ptr = libSingular.n_Init(0, parent(x).ptr)
+   libSingular.n_Delete(x.ptr, parent(x).ptr)
+   x.ptr = ptr
+   return x
+end
+
+###############################################################################
+#
+#   Conversions and promotions
+#
+###############################################################################
+
+promote_rule(C::Type{n_transExt}, ::Type{T}) where T <: Integer = n_transExt
+
+promote_rule(C::Type{n_transExt}, ::Type{n_Z}) = n_transExt
+
+###############################################################################
+#
+#   Parent call functions
+#
+###############################################################################
+
+function (R::N_FField)()
+   z = n_transExt(R)
+   z.parent = R
+   return z
+end
+
+function (R::N_FField)(x::Integer)
+   z = R(libSingular.n_InitMPZ(BigInt(x), R.ptr)) 
+   z.parent = R
+   return z
+end
+
+function (R::N_FField)(n::Int)
+   z = n_transExt(R, n)
+   z.parent = R
+   return z
+end
+
+(R::N_FField)(n::n_transExt) = n
```

Comment:
Should probably check the parent is correct.

Reply:
I fixed it.